src/networking.c

```diff
@@ -1775,8 +1777,17 @@ int freeClientsInAsyncFreeQueue(void) {
     while ((ln = listNext(&li)) != NULL) {
         client *c = listNodeValue(ln);
 
-        if (c->flags & CLIENT_PROTECTED) continue;
-
+        if (c->flags & CLIENT_PROTECTED) {
+            if (!(c->flags & CLIENT_REPL_RDB_CHANNEL)) continue;
+            /* Check if we can remove RDB connection protection. */
+            if (!c->first_free_time) {
+                c->first_free_time = server.unixtime;
+                continue;
+            }
+            if (server.unixtime - c->first_free_time > WAIT_BEFORE_RDB_CLIENT_FREE) {
```

Comment:
here it looks clear that `first_free_time` should be renamed to include the words RDB_CLIENT